composites/Plugin/SocialPreviews/Facebook/components/FacebookImage.js

```diff
@@ -0,0 +1,154 @@
+	height: ${ props => props.dimensions.height };
+	width: ${ props => props.dimensions.width };
+	display: flex;
+	justify-content: center;
+`;
+
+const ErrorImage = styled.div`
+	height: 261px;
+	width: 500px;
+	text-align: center;
+	box-sizing: border-box;
+	padding-top: 7em;
+	font-size: 1rem;
+	color: ${colors.$color_white};
+	background-color: ${colors.$color_red};
+`;
+
+const PlaceholderImage = styled.div`
+	height: 261px;
+	width: 500px;
+	background-color: ${colors.$color_grey};
+`;
+
+/**
+ * Renders the FacebookImage component.
+ *
+ * @param {string} src The image source.
+ *
+ * @returns {ReactComponent} The FacebookImage component.
+ */
+export default class FacebookImage extends React.Component {
+	/**
+	 * The constructor.
+	 *
+	 * @param {Object} props The component's props.
+	 */
+	constructor( props ) {
+		super( props );
+		this.state = {
+			imageProperties: null,
+			status: "loading",
+		};
+	}
+
+	/**
+	 * After the component did mount, determine the properties of the FacebookImage.
+	 *
+	 * @returns {Promise} Resolves when there are image properties.
+	 */
+	componentDidMount() {
+		return determineFacebookImageProperties( this.props.src ).then( ( imageProperties ) => {
+			this.setState( {
+				imageProperties: imageProperties,
+				status: "loaded",
+			} );
+		} ).catch( () => {
+			this.setState( {
+				imageProperties: null,
+				status: "errored",
+			} );
+			return true;
+		} );
+	}
+
+	/**
+	 * Gets the dimensions for the Facebook image container.
+	 *
+	 * @param {string} imageMode The facebook image mode: either landscape, square or portrait.
+	 *
+	 * @returns {Object} The width and height for the container.
+	 */
+	getContainerDimensions( imageMode ) {
+		switch ( imageMode ) {
+			case "square":
+				return {
+					height: SQUARE_HEIGHT + "px",
+					width: SQUARE_WIDTH + "px",
+				};
+			case "portrait":
+				return {
+					height: PORTRAIT_HEIGHT + "px",
+					width: PORTRAIT_WIDTH + "px",
+				};
+			case "landscape":
+			default:
+				return {
+					height: LANDSCAPE_HEIGHT + "px",
+					width: LANDSCAPE_WIDTH + "px",
+				};
+		}
+	}
+
+	/**
+	 * Renders the FacebookImage.
+	 *
+	 * @returns {ReactComponent} Either the ErrorImage component or the FacebookImageContainer.
+	 */
+	render() {
+		const imageProperties = this.state.imageProperties;
+		const status = this.state.status;
+
+		if ( status === "loading" ) {
+			return <PlaceholderImage />;
+		}
+
+		if ( status === "errored" ) {
+			return <ErrorImage>The given image url cannot be loaded</ErrorImage>;
+		}
+		if ( imageProperties.height < MIN_IMAGE_HEIGHT || imageProperties.width < MIN_IMAGE_WIDTH ) {
+			return <ErrorImage>The image you selected is too small for Facebook</ErrorImage>;
+		}
+
+		const containerDimensions = this.getContainerDimensions( imageProperties.mode );
+		return <FacebookImageContainer mode={ imageProperties.mode } dimensions={ containerDimensions }>
+			<StyledImage src={ this.props.src } mode={ imageProperties.mode } />
```

Comment:
- The property `mode` gets rendered in the HTML as an attribute because React thinks it's a valid HTML attribute: it should be renamed.
- the image misses an `alt` attribute: more on this later